src/components/Checkout/InContext/InContextCheckout.vue

```diff
@@ -69,21 +69,68 @@
 				{{ customCheckoutButtonText }}
 			</kv-button>
 		</div>
-		<div class="in-context-payment-conttrols" v-else>
+		<div class="in-context-payment-controls" v-else>
+			<kv-grid v-if="!continueAsGuest" class="tw-grid-cols-6">
+				<div class="tw-col-start-4 tw-col-end-7 tw-text-right">
+					<kv-button
+						class="smaller checkout-button tw-mb-1"
+						v-if="promoGuestCheckoutEnabled"
+						@click="continueAsGuest = true"
+					>
+						Continue as a guest
+					</kv-button>
+					<kv-button
+						class="smaller checkout-button tw-mb-1"
+						v-if="promoGuestCheckoutEnabled"
+						variant="secondary"
+						:href="registerOrLoginHref"
+					>
+						Continue as an existing user
+					</kv-button>
+				</div>
+				<div class="tw-col-start-4 tw-col-end-7">
+					<p class="tw-grid-row">
+						For existing users, we recommend you log in with your work email address.
+					</p>
+				</div>
+			</kv-grid>
+
 			<kiva-credit-payment
-				v-if="showKivaCreditButton"
+				v-if="showKivaCreditButton && !promoGuestCheckoutEnabled"
 				@complete-transaction="completeTransaction"
 				class="checkout-button"
 				id="kiva-credit-payment-button"
-				@refreshtotals="$emit('refreshtotals')"
+				@refreshtotals="$emit('refresh-totals')"
+				@updating-totals="setUpdatingTotals"
+				@checkout-failure="handleCheckoutFailure"
+			/>
+			<kiva-credit-guest-payment
+				v-if="showKivaCreditButton
+					&& promoGuestCheckoutEnabled
+					&& continueAsGuest"
+				:is-guest-checkout="true"
+				@complete-transaction="completeTransaction"
+				class="checkout-button"
+				id="kiva-credit-payment-button"
+				:promo-fund-id="String(promoFundId)"
+				:managed-account-id="managedAccountId"
+				:promo-guest-checkout-enabled="promoGuestCheckoutEnabled"
+				:promo-name="promoName"
+				@refreshtotals="$emit('refresh-totals')"
 				@updating-totals="setUpdatingTotals"
 				@checkout-failure="handleCheckoutFailure"
 			/>
 
 			<checkout-drop-in-payment-wrapper
-				v-else
+				v-if="!showKivaCreditButton
+					&& continueAsGuest"
```

Comment:
This shouldn't be gated on `continueAsGuest`. With this in place the Braintree drop-in interface will now show up with payment is needed, blocks checkout.

Reply:
We gate all payment methods with a button to select which payment method is shown.  In the case that "continue as a guest" is shown, THEN we show the checkout-drop-in-payment wrapper.  That's what this flag does.

Reply:
It doesn't mean that there's a case where the drop in payment wrapper is inaccessible.